salt/common/files/log-rotate.conf

```diff
@@ -11,13 +11,13 @@
 /opt/so/log/soctopus/*.log
 /opt/so/log/curator/*.log
 /opt/so/log/fleet/*.log
-/opt/so/log/suricata/*.log
+/opt/so/log/suricata/suricata.log
 /opt/so/log/mysql/*.log
 /opt/so/log/playbook/*.log
-/opt/so/log/logstash/*.log
-/opt/so/log/filebeat/*.log
 /opt/so/log/telegraf/*.log
 /opt/so/log/redis/*.log
+/opt/so/log/sensoroni/*.log
+/opt/so/log/stenographer/*.log
```

Comment:
Some of my rigs are in a same situation as `[suriloss.sh] <defunct>` that I've previously mentioned when the log size becomes large (around 100MB).
https://www.reddit.com/r/securityonion/comments/iaor6m/20_lots_of_defunct_surilosssh/

So I don't want to accept this suggestion at this moment.
But this is not my repository and you really need to remove the line in order to investigate a bug, I will obey.

Reply:
I've found causes of defunct processes, but this problem is little bit complex.
I'm now sure that the log file size is one of causes.
I will explain others later...

Reply:
If this bug is directly caused by the log file's size growing too large then I don't really have any problem rotating it. I'd be very interested to hear the other causes you've found, I wasn't involved in the earlier fix so I'm not familiar with what's happening here.

Reply:
A cause of defunct processes is docker's PID 1 zombie reaping problem.
https://blog.phusion.nl/2015/01/20/docker-and-the-pid-1-zombie-reaping-problem/
To avoid this, add "--init" option into docker run command. For SO2.3.21, add `    - init: True` around line 45 of `salt/telegraf/init.sls` and restart telegraf. If you did correctly, you can see the difference between PID 1 with/without the option by `docker exec so-telegraf ps -ef` command.

But this is just avoiding to produce defunct processes and not a root cause.

It is a timeout value of telegraf config.
Line 730 of `salt/telegraf/etc/telegraf.conf` is commented out, so the timeout value for shell commands such as `suriloss.sh` is 5 seconds as default. Usually it's enough to process the log; however, it takes longer to read log files when the they are large. Moreover the shells run in parallel, so the disk I/O rate becomes higher and reading the (large) log files are slower even if using `tac` command or `grep`'s `m` option. As a result, processes are timed out and killed by telegraf, then docker's PID 1 problem happens.
Aborting processes may give inaccurate results of log parsing. (The process are usually terminated at tac and grep commands during my investigation)

In conlusion, to solve this problem we need:
1. to use faster disk devices
2. to keep log files small
3. to tune telegraf more
4. then, add init option